test_generators/performance_suite_generator.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+"""
+Script to generate a TestSuite of TestCases from specified TestAssets
+"""
+import copy
+import glob
+import httpx
+import io
+import json
+import logging
+from pathlib import Path
+import tempfile
+import zipfile
+
+from translator_testing_model.datamodel.pydanticmodel import (
+    TestSuite,
+    TestAsset,
+    TestMetadata,
+    TestPersonaEnum,
+    TestSourceEnum,
+    TestObjectiveEnum,
+    TestEnvEnum,
+)
+
+from test_generators.utils import create_test_cases_from_test_assets, dump_to_json
+
+
+def generate_message(query, num_curies):
+    nodes = query["message"]["query_graph"]["nodes"]
+    for node in nodes.values():
+        if "ids" in node:
+            node["ids"] = node["ids"][:num_curies]
+    query["bypass_cache"] = True
+    return query
+
+
+creative_query = {
+    "message": {
+        "query_graph": {
+            "nodes": {
+                "ON": {
+                    "categories": [
+                        "biolink:Disease"
+                    ],
+                    "ids": [
+                        "MONDO:0005301"
+                    ]
+                },
+                "SN": {
+                    "categories": [
+                        "biolink:ChemicalEntity"
+                    ]
+                }
+            },
+            "edges": {
+                "t_edge": {
+                    "object": "ON",
+                    "subject": "SN",
+                    "predicates": [
+                        "biolink:treats"
+                    ],
+                    "knowledge_type": "inferred"
+                }
+            }
+        }
+    },
+    "bypass_cache": True,
+}
+
+
+def create_test_suite(
+    logger: logging.Logger,
+) -> None:
+    """Download tests from specified location."""
+    with open("./asset_backups/kp_performance_tests_2024_10_18.json", "r") as f:
+        data = json.load(f)
+
+        test_suite = {
+            "id": "performance_tests",
+            "name": "Performance Tests",
+            "description": "Performance Tests",
+            "tags": [],
+            "test_runner_settings": [],
+            "test_metadata": {},
+            "test_persona": "Developer",
+            "test_suite_specification": None,
+            "test_cases": {},
+        }
+        # sequential KP
+        for rate in [(900, 1), (180, 10), (90, 100), (30, 1000)]:
```

Comment:
maybe for next iteration we could pull these values out into named parameters and/or a config yaml?

Reply:
Great idea!